### PR TITLE
don't print the duration in quiet mode

### LINKED ISSUE #816: Hide test duration in `--quiet` mode
This just occurred to me.

At the very end of Cucumber's pretty output, there's a line that shows the duration of the test run.

e.g.
```
➤ cucumber features/docs/getting_started.feature -f pretty
Using the default profile...
Feature: Getting started

  To get started, just open a command prompt in an empty directory and run
  `cucumber`. You'll be prompted for what to do next.

  @spawn
  Scenario: Run Cucumber in an empty directory                              # features/docs/getting_started.feature:7
    Given a directory without standard Cucumber project directory structure # features/lib/step_definitions/cucumber_steps.rb:1
    When I run `cucumber`                                                   # aruba-0.6.1/lib/aruba/cucumber.rb:93
    Then it should fail with:                                               # aruba-0.6.1/lib/aruba/cucumber.rb:205
      """
      No such file or directory - features. You can use `cucumber --init` to get started.
      """

  Scenario: Accidentally run Cucumber in a folder with Ruby files in it.    # features/docs/getting_started.feature:15
    Given a directory without standard Cucumber project directory structure # features/lib/step_definitions/cucumber_steps.rb:1
    And a file named "should_not_load.rb" with:                             # aruba-0.6.1/lib/aruba/cucumber.rb:29
      """
      puts 'this will not be shown'
      """
    When I run `cucumber`                                                   # aruba-0.6.1/lib/aruba/cucumber.rb:93
    Then the exit status should be 1                                        # aruba-0.6.1/lib/aruba/cucumber.rb:197
    And the output should not contain:                                      # aruba-0.6.1/lib/aruba/cucumber.rb:160
      """
      this will not be shown

      """

2 scenarios (2 passed)
8 steps (8 passed)
0m0.641s
```

That `0m0.641s` is what I propose to remove in quiet mode.

It would mean we can use `Then it should pass with exactly:` in our tests, which gives much better feedback than `Then it should pass with:` and not need [this hack](https://github.com/cucumber/cucumber/blob/master/features/lib/support/normalise_output.rb#L11).

### PR DESCRIPTION
**Pull request by richarda**
This attempts to resolve #816 
I'm not able to remove the [referenced hack](https://github.com/cucumber/cucumber/blob/master/features/lib/support/normalise_output.rb#L11) just yet, but I'll keep working on it.

### CODE REVIEW

**Comment by richarda** [comment]
@mattwynne does this do it for the Tcl situation or do we need to be able to remove the duration from `--dry-run` features too?

File: lib/cucumber/formatter/console.rb [hunk 1]
```diff
@@ -85,8 +85,8 @@ def print_stats(features, options)
         @io.puts scenario_summary(runtime) {|status_count, status| format_string(status_count, status)}
         @io.puts step_summary(runtime) {|status_count, status| format_string(status_count, status)}
 
-        @io.puts(format_duration(features.duration)) if features && features.duration
-
+        @io.puts(format_duration(features.duration)) if features && features.duration && options[:duration]
```
Comment (mattwynne) [inline lib/cucumber/formatter/console.rb]: I wonder if we still need that `features.duration` check. Could it ever be nil? Looks quite defensive.

**Comment by mattwynne** [comment]
LGTM @richarda. How come you can't remove the hack?

**Comment by richarda** [comment]
@mattwynne in the current state of this branch, removing the line that substitutes a fixed duration of 0m0.012 leaves a handful of scenarios failing with what seems to be their actual duration.  I was just still thinking what might be a good way to address that.  The failing scenarios are using `Then it should pass/fail with exactly`  Do you think some of those steps might not actually require an exact output comparison if the purpose involves only a portion of the output?  Should `dry-run` print a duration?

These are are the tests that break without that substitution:
```
Failing Scenarios:
cucumber -p ruby_2_0 features/docs/cli/dry_run.feature:8 # Scenario: With a failing step
cucumber -p ruby_2_0 features/docs/cli/dry_run.feature:30 # Scenario: In strict mode
cucumber -p ruby_2_0 features/docs/cli/dry_run.feature:52 # Scenario: In strict mode with an undefined step
cucumber -p ruby_2_0 features/docs/defining_steps/table_diffing.feature:7 # Scenario: Extra row
cucumber -p ruby_2_0 features/docs/exception_in_before_hook.feature:44 # Scenario: Handle Exception in Before hook for Scenario with Background
cucumber -p ruby_2_0 features/docs/formatters/usage_formatter.feature:60 # Scenario: Run with --expand --format usage
cucumber -p ruby_2_0 features/docs/formatters/usage_formatter.feature:86 # Scenario: Run with --format stepdefs
cucumber -p ruby_2_0 features/docs/wire_protocol/table_diffing.feature:99 # Scenario: Invoke a step definition which asks for an immediate diff that fails
cucumber -p ruby_2_0 features/docs/writing_support_code/tagged_hooks.feature:34 # Scenario: omit tagged hook
cucumber -p ruby_2_0 features/docs/writing_support_code/tagged_hooks.feature:54 # Scenario: omit tagged hook
cucumber -p ruby_2_0 features/docs/writing_support_code/tagged_hooks.feature:69 # Scenario: Omit example hook
```

**Comment by mattwynne** [comment]
You're right actually, it seems nonsensical to get the duration of a dry-run, so I think it would be OK for that to imply quiet.

**Comment by mattwynne** [comment]
Can you review the other ones to see whether it would make sense just to add `--quiet` to the command run on the `When` step? A quick scan over them looks like that would be OK.

**Comment by richarda** [comment]
Yes.  I'll review those scenarios and take a look at dry-run.

**Comment by brasmusson** [comment]
@mattwynne You said yourself that  `Then it should pass with exactly:` is preferable to `Then it should pass with:` since it gives much better feedback. If/When defining a feature to check the indentation for source references for Scenario Outlines in expand mode ([where we have known problems](https://github.com/cucumber/cucumber/commit/84eb09c395b5ad3b33078bef66ef16cdfa476af8#commitcomment-7362057)), `--quiet` cannot be used, and the it is only "the hack" that enables the use of `Then it should pass with exactly:`.

If you really want to remove "the hack", then a `--no-duration` option is needed, so the source and/or snippets still can be checked with `Then it should pass with exactly:`.

**Comment by mattwynne** [comment]
Sounds like @brasmusson has thought about this more than me, and we should have a separate `--no-duration` flag that's implied by `--quiet`. Right?

**Comment by richarda** [comment]
@mattwynne when I ran the build locally against the latest `cucumber-ruby-core` from github, I got similar failures to what Travis is reporting on the PR.  When I ran it against the released version of the core, tests pass.  Is that true for you?

**Comment by richarda** [comment]
I'm good with `--no-duration` implied by `--quiet`.  I also like `--dry-run` implying `--no-duration` as well.

**Comment by mattwynne** [comment]
Has it really been since March?! I've been too busy.

Thanks for your contribution @richarda, sorry to have stalled on this for so long.

**Comment by richarda** [comment]
Ha!  No problem.  I was just here looking at open issues again too.

**Pull request merged**